### ISSUE TITLE
Request: Resume possibility

### ISSUE THREAD

**Issue by SeriousGeorge** [Microsoft/WindowsUIDevLabs#169]
Hi there,

I understand that a pause method from animations was removed and instead we have to define the stopbehaviour (like AnimationStopBehavior.LeaveCurrentValue).

But how do we resume?
It would be great to have a resume function or at least something like AnimationStartBehaviour.

background:
I've a list that keeps track of all animations ongoing on screen.
On pausing the app (e.g. pressing space) I iterate through the list and call .stop on all animations.
Now I'd like to resume all animations from the current state if the user is pressing space again.

Best regards

**Comment by patsiu** [comment]
Hey @SeriousGeorge,

Yes, like you said you would stop the animation and define a StopBehavior. To "resume" you would have to start a new animation on the Visual. The new animation could use an ExpressionKeyFrame with "this.StartingValue" to reference its current value.

That said, we are thinking about how we want to bring the playback controls for KeyFrameAnimations for a future release.

**Comment by SeriousGeorge** [comment]
OK. Phew. But than I have to keep track all properties that where animated in order to construct a new animation? Or can I clone the stopped animation object and change the starting keyframe to "this.StartingValue" as you suggested?

**Comment by patsiu** [comment]
Yes, unfortunately in the current model you have to do a bit of book keeping to know which Visuals and their corresponding properties were stopped in order to know who to start up again. That said, think of the Composition Animations you write in code as templates - you can reference the same animation in different StartAnimation() calls that target different Visuals. When you call StartAnimation, you create an "instance" of that template. 

Using "this.StartingValue" will definitely help here. And assuming you stop the first instance to create a "pause" behavior before the KFA reaches the end and the end value doesn't change, you can most likely use the same template when you call StartAnimation() again.

**Comment by daneuber** [comment]
Thanks for the feedback. Please feel free to file a feature request in the backlog [here ](https://developer.microsoft.com/en-us/windows/platform/features) so we can prioritize this among our other backlog items.

**Comment by SeriousGeorge** [comment]
Hi there,

meanwhile I switched my whole animation code from storyboards to ScalarKeyFrameAnimations.
I implemented pause/resume as suggested above. As long as this basically works (setting up the animations as templates and reuse them, combined with the startingValue Property) it has some pitfalls I think.

a.) If paused and resumed the animation would play wirh the initial duration again, adding up to the total duration.

For example... if having an animation with duration 10 seconds, pausing at 9 seconds and then resuming, it would resume with the correct animation properties, but adding again 10 secodns, resulting in 19s.

b.) There is - at least I seen none - no way to get the current duration or playhead position so to speak.
If there is such a method to extract the current time of the animation, we could at least store that as a workaround to resume at the correct time again.

This is a big problem to me, as I'm trying to make an offscreen renderer with win2d combined with UI-Composition to render UI-Animations as Imagesequences to a video. Plus since my current app in the store  is a Slideshowapp, pausing is not working as expected.
